testing/scenario_app/android/app/src/androidTest/java/dev/flutter/TestRunner.java

```diff
@@ -5,18 +5,19 @@
 package dev.flutter;
 
 import android.os.Bundle;
+import androidx.annotation.Nullable;
 import androidx.test.runner.AndroidJUnitRunner;
 import dev.flutter.scenariosui.ScreenshotUtil;
 
 public class TestRunner extends AndroidJUnitRunner {
   @Override
-  public void onCreate(Bundle arguments) {
+  public void onCreate(@Nullable Bundle arguments) {
     ScreenshotUtil.onCreate(this, arguments);
     super.onCreate(arguments);
   }
 
   @Override
-  public void finish(int resultCode, Bundle results) {
+  public void finish(int resultCode, @Nullable Bundle results) {
```

Comment:
are we getting lint warnings for `@Nullable/@NonNullable`?

Reply:
Yes

Reply:
We should enable these for the embedding too. I suspect there are places where we don't have the right annotations.

Reply:
Would it be too crazy if we remove flutter.jar and instead link the source directly?

Reply:
We have a seprate script for linting the embedding at `//android_lint`. There's a baseline file in there with things we already are failing on. That could be driven down separately.

I think this is a better test if we link in flutter.jar the way regular applications do - not quite sure what you have in mind here though.

Reply:
This way we don't need https://github.com/flutter/engine/blob/1eb8a34a3975070c42bc69f3981348dfa4f50a2d/tools/android_lint/bin/main.dart, right?

Since the linter is usually called by AGP, I wonder if there are additional benefits that required a more conventional Android project structure

Reply:
We don't have a normal Gradle project there though.

Reply:
it still builds with Gradle and AGP right? or is build.gradle  never used?

Reply:
That tool directly invokes lint. It does not use Gradle or agp.

Reply:
The scenario app itself is a Gradle project, no? I was thinking since it is, it can also lint the embedding source code